src/main/java/github/update/PageHeaderIterator.java

```diff
@@ -0,0 +1,137 @@
+package github.update;
+
+import static org.eclipse.egit.github.core.client.IGitHubConstants.PARAM_PAGE;
+import github.GitHubClientExtended;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.egit.github.core.client.GitHubRequest;
+import org.eclipse.egit.github.core.client.GitHubResponse;
+import org.eclipse.egit.github.core.client.NoSuchPageException;
+import org.eclipse.egit.github.core.util.UrlUtils;
+
+import util.HTLog;
+
+/**
+ * Iterator for getting values of a header field from paged responses
+ * @see org.eclipse.egit.github.core.client.PageIterator
+ */
+public class PageHeaderIterator implements Iterable<String>, Iterator<String> {
+    private static final Logger logger = HTLog.get(PageHeaderIterator.class);
+
+    private final GitHubRequest request;
+    private final GitHubClientExtended client;
+    private final String headerField;
+
+    private int nextPage; // Current page number
+    private String next; // Next uri to be fetched
+    private HttpURLConnection lastConnection;
+
+    public PageHeaderIterator(GitHubRequest request, GitHubClientExtended client,
+                              String headerField) {
+        this.request = request;
+        this.client = client;
+        this.headerField = headerField;
+
+        next = request.getUri();
+        nextPage = parsePageNumber(next);
+    }
+
+    public GitHubRequest getRequest() {
+        return request;
+    }
+
+    public HttpURLConnection getLastConnection() {
+        return lastConnection;
+    }
+
+    /**
+     * Parses page number from uri
+     *
+     * @param uri
+     * @return page number
+     */
+    private int parsePageNumber(String uri) {
+        if (uri == null || uri.length() == 0) {
+            return -1;
+        }
+
+        final URI parsed;
+        try {
+            parsed = new URI(uri);
+        } catch (URISyntaxException e) {
+            return -1;
+        }
+
+        final String param = UrlUtils.getParam(parsed, PARAM_PAGE);
+        if (param == null || param.length() == 0) {
+            return -1;
+        }
+
+        try {
+            return Integer.parseInt(param);
+        } catch (NumberFormatException nfe) {
+            return -1;
+        }
```

Comment:
This would be a great place to use the Maybe monad. Unfortunately lambdas that throw exceptions need a whole new set of functional interfaces... I guess it isn't worth it? :disappointed: